ouroboros-consensus-test/test-storage/Test/Ouroboros/Storage/ChainDB/Model.hs

```diff
@@ -876,8 +877,11 @@ validChains cfg m bs =
     -- first, which results in the valid chain A->B', which is then chosen
     -- over the equally preferable A->B as it will be the first in the list
     -- after a stable sort.
-    sortChains $ chains bs
+    sortChains $ pruneKnownInvalid (invalid m) <$> chains bs
   where
+    pruneKnownInvalid invalidBlocks chain =
+      Chain.takeWhileChain (\b -> blockHash b `Map.notMember` invalidBlocks) chain
```

Comment:
Let's include a comment why we do this, sth along the lines of

> We don't want known-invalid blocks to influence the sorting of candidate chains.

Also a small style thing: What about making `invalidBlocks` a `where` binding under `pruneKnownInvalid`?